### PR TITLE
Desugared Daml golden tests

### PR DESCRIPTION
**Pull request by akrmn**

### CODE REVIEW

File: compiler/damlc/tests/daml-test-files/ExceptionCatchError.EXPECTED.desugared-daml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+test1
+  = scenario
+      do p <- getParty "Alice"
+         submit
+           p
+           do DA.Internal.Desugar._tryCatch
+                \ _ -> do error "hello"
+                \case
+                  DA.Internal.Desugar.fromAnyException -> DA.Internal.Desugar.Some (GeneralError msg)
+                    -> DA.Internal.Desugar.Some msg === "hello"
```
Comment (sofiafaro-da) [inline compiler/damlc/tests/daml-test-files/ExceptionCatchError.EXPECTED.desugared-daml:17]: Oh ... this is confusing. `DA.Internal.Desugar.fromAnyException -> DA.Internal.Desugar.Some (GeneralError msg)` is a view pattern! We should really start adding parentheses everywhere in our desugarings. But this is unrelated to this PR.

**Review by sofiafaro-da — APPROVED** [review]
Looks great, thank you!

Do these update like the daml-doc golden tests? Can you add a section in the `compiler/damlc/README.md` file about how to update the desugar golden tests? Thank you.

**Comment by akrmn** [comment]
> Do these update like the daml-doc golden tests? Can you add a section in the `compiler/damlc/README.md` file about how to update the desugar golden tests? Thank you.

Yes they update just like the `daml-doc` tests, with `--accept`. No special logic was needed, `tasty-golden` takes care of the flag on its own. I've added two sections in `compiler/damlc/README.md` - one for the test and another for the library that provides the `desugar` function.
